include/deal.II/numerics/cell_data_transfer.templates.h

```diff
@@ -0,0 +1,196 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_cell_data_transfer_templates_h
+#define dealii_cell_data_transfer_templates_h
+
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/numerics/cell_data_transfer.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+
+namespace
+{
+  template <typename VectorType>
+  void
+  post_unpack_action(VectorType &out)
+  {
+    out.compress(::dealii::VectorOperation::insert);
+  }
+
+  template <typename value_type>
+  void
+  post_unpack_action(std::vector<value_type> &)
+  {
+    // Do nothing for std::vector as VectorType.
+  }
+} // namespace
```

Comment:
Having anonymous namespaces in headers is a bit risky: I recommend putting these in `namespace internal` instead.